stdlib/map.ml

```diff
@@ -425,6 +426,18 @@ module Make(Ord: OrderedType) = struct
           if pvd then if l==l' && r==r' then m else join l' v d r'
           else concat l' r'
 
+    let rec filter_map f = function
+        Empty -> Empty
+      | Node {l; v; d; r} ->
+          (* call [f] in the expected left-to-right order *)
+          let l' = filter_map f l in
+          let pvd = f v d in
```

Comment:
This variable should probably be called `fvd`. (Apologies for extreme nitpicking!)

Reply:
Done. The follow-up question would be: why `v` and `d`? Who knows.